lib/matplotlib/lines.py

```diff
@@ -657,7 +657,7 @@ def set_data(self, *args):
         *args : (N, 2) array or two 1D arrays
         """
         if len(args) == 1:
-            x, y = args[0]
+            (x, y), = args
```

Comment:
Not sure if that's really better in this context.

Reply:
I prefer this form as it's immediately clear that you're not dropping anything from args, i.e. that single line also asserts that args has length 1.  (Yes, it's redundant with the line just above, but I still think it helps the readability.)